File: algorithms/python/opposite_sign_elements.py
numbers = [1, -2, 3, 4, 5, -6]
countOppositeSignElements = 0

i = 0
j = 1
while i < len(numbers)-1:
    if (numbers[i] > 0 and numbers[j] < 0) or (numbers[i] < 0 and numbers[j] > 0):
        countOppositeSignElements += 1
    
    i += 1
    j += 1

print("Count of opposite sign elements:", countOppositeSignElements)

File: algorithms/cpp/opposite_sign_elements.cpp
#include <iostream>
using namespace std;

int main() 
{
    int numbers[] = {1, -2, 3, 4, 5, -6};
    int length = 6;
    int oppositeSignElementsCount = 0;
    
    int j = 1;
    for( int i=0; i < length-1; i++ )
    {
        if( ( numbers[i] > 0 && numbers[j] < 0 ) || ( numbers[i] < 0 && numbers[j] > 0 ) )
        {
            oppositeSignElementsCount++;
        }

        j++;
    }

    cout << "opposite sign elements count: " << oppositeSignElementsCount << "\n";

    return 0;
}
